### PR TITLE
Client demo

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

**Comment by alicoding** [comment]
User bower version of ace

**Comment by alicoding** [comment]
This is hard to review... I think it will be better if you could clean up code for those files that we will keep (not demo part).

File: client/demo/index.js [hunk 1]
```diff
@@ -1,202 +1,87 @@
...
-      text('#checksumsGet', 'Unable to receive checksums', true);
+var count = 1;
+var files = [];
+var currentPath;
+$(document).ready(function() {
+  var editor = ace.edit(document.getElementById('editor'));
+  editor.setTheme("ace/theme/monokai");
+  editor.getSession().setMode("ace/mode/javascript");
+  var Filer = makedrive.Filer;
+  var fs = new Filer.FileSystem({provider: new Filer.FileSystem.providers.Memory("B")});
```
Comment (sedge) [inline client/demo/index.js:9]: "B"?

File: client/demo/index.js [hunk 2]
```diff
@@ -1,202 +1,87 @@
...
+    var uri = 'http://localhost:9090';
+    makedrive.init(uri, fs, function(err) {
+      if(err) {
+        return console.log(err);
+      }
+      fs.watch('/projects', {recursive: true}, function(event, filename) {
+        var found = false;
+        var fname = filename.replace(/^\//, "");
+        fname = Filer.Path.basename(fname);
+        console.log('Change to:', fname);
```
Comment (sedge) [inline client/demo/index.js:24]: Let's delete the `console.log`s

File: client/demo/makedrive-demo.js [hunk 3]
```diff
@@ -1,6 +1,12 @@
 var fs = new Filer.FileSystem({
   provider: new Filer.FileSystem.providers.Memory()
 });
+var makedriveOptions = {
```
Comment (sedge) [inline client/demo/makedrive-demo.js:4]: Is this file even being used?

File: client/src/client-server-sync.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    rsyncOptions = {
+      size: 5,
+      time: true,
+      recursive: true
+    };
+
+module.exports = {
+  sync: function sync( fs, uri, connectionId, path, callback ) {
+    uri = uri + '/api/sync/';
+    var reqOptions = {
```
Comment (sedge) [inline client/src/client-server-sync.js:15]:
``` javascript
var reqOptions = {
      uri: uri + connectionId
    },
    syncID
```

File: client/src/client-server-sync.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            if( error || response.statusCode !== 200 ) {
+              return callback( error );
+            }
+            var checksums = JSON.parse( body ).checksums;
+            console.log(checksums);
+            rsync.diff( fs, path, checksums, rsyncOptions, function( err, diffs ) {
+              if( err ) {
+                return callback( err );
+              }
+              console.log('Diffs;', diffs);
```
Comment (sedge) [inline client/src/client-server-sync.js]: delete this line

File: client/demo/index.js [hunk 6]
```diff
@@ -1,202 +1,87 @@
...
-                            }).done(f);
-                          }
-                        });
-                      });
-                    }
-                  });
-                }
-              });
+      $('.file').unbind('click').click( function(e) {
+        console.log(e.target.innerHTML);
```
Comment (sedge) [inline client/demo/index.js:54]: delete

File: client/src/client-server-sync.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+          }
+          reqOptions.uri = uri + 'checksums';
+          reqOptions.method = 'GET';
+          delete reqOptions.json;
+          request( reqOptions, function( error, response, body ) {
+            if( error || response.statusCode !== 200 ) {
+              return callback( error );
+            }
+            var checksums = JSON.parse( body ).checksums;
+            console.log(checksums);
```
Comment (sedge) [inline client/src/client-server-sync.js]: delete

File: client/src/client-server-sync.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+                      }
+                    }
+                  }
+                } else {
+                  for ( k = 0; k < diffs[i].diffs.length; k++ ) {
+                    if ( diffs[i].diffs[k].data ) {
+                      req.attach( 'webmasterfile', new Blob( [diffs[i].diffs[k].data], {
+                        type: 'application/octet-binary'
+                      } ), diffs[i].diffs[k].path );
+                      console.log(diffs[0].diffs);
```
Comment (sedge) [inline client/src/client-server-sync.js]: delete

File: client/src/comms.js [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+        });
+        socket.send(data);
+        // Set up listeners
+        socket.onmessage = function b( data, flags ) {
+          // Parse the data into an object
+          data = data.data;
+          try {
+            data = JSON.parse( data );
+          } catch( e ) {
+            console.log('1', e);
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+            return callback( e );
+          }
+          
+          if( data.type === SyncMessage.RESPONSE && data.name === SyncMessage.ACK ) {
+            syncState = states.CONN_OPEN;
+            socket.removeEventListener('message', b);
+            socket.onmessage = function(data, flags) {
+              data = data.data;
+              console.log('Data:', data);
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+              try {
+                data = JSON.parse( data );
+              } catch( e ) {
+                console.log('2', e);
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+                return callback( e );
+              }
+              // Is the object a request or response
+              if ( data.type === SyncMessage.RESPONSE ) {
+                console.log('State:', syncState);
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+                if ( data.name === SyncMessage.SOURCE_LIST  && syncState === states.CONN_OPEN ) {
+                  var srcList = data.content.srcList;
+                  path = data.content.path;
+
+                  return rsync.checksums( fs, path, srcList, rsyncOptions, function( err, checksums ) {
+                    if ( err ) {
+                      console.log('3', err);
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+
+                if( data.name === SyncMessage.DIFF && states.CHKSUM_ACK ) {
+                  var diffs = data.content.diffs;
+                  path = data.content.path;
+
+                  convert.call( diffs );
+
+                  return rsync.patch( fs, path, diffs, rsyncOptions, function( err ) {
+                    if ( err ) {
+                      console.log('4', err);
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+                  syncState = states.CHKSUM_ACK;
+                  var message = new SyncMessage( SyncMessage.REQUEST, SyncMessage.DIFF );
+                  message.setContent( { checksums: checksum } );
+                  socket.send( JSON.stringify( message ) );
+                  return callback();
+                }
+
+                if( data.name === SyncMessage.ERROR ) {
+                  syncState = states.ERROR;
+                  console.log(data);
```
Comment (sedge) [inline client/src/comms.js]: delete `171` and `172`

File: client/src/comms.js [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+                  console.log('5');
+                  return callback( new Error( data.content ) );
+                }
+                
+                if( data.name === SyncMessage.ACK ) {
+                  syncState = states.CONN_OPEN;
+                  return callback();
+                }
+                console.log('6');
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+                return callback( new Error( data.content ) );
+              }
+              console.log('7');
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+              return callback(new Error('Cannot handle message'));
+            };
+            return initial(null, connectionId);
+          }
+          console.log('8');
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 19]
```diff
@@ -0,0 +1,205 @@
...
+          return initial(new Error('Cannot connect to server'));
+        };
+      };
+
+      socket.onclose = function() { 
+        syncState = states.CONN_CLOSED;
+        console.log('10');
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/comms.js [hunk 20]
```diff
@@ -0,0 +1,205 @@
...
+        return callback( new Error( 'Socket connection was closed' ));
+      };
+
+      socket.onerror = function( err ) {
+        console.log('9', err);
```
Comment (sedge) [inline client/src/comms.js]: delete

File: client/src/interface.js [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+  serverURI = uri;
+  var options = {
+    uri: serverURI,
+    fs: fs
+  };
+  initialize(options, function(err, id) {
+    if(err) {
+      return callback(err);
+    }
+    console.log('Id:', id);
```
Comment (sedge) [inline client/src/interface.js]: delete

File: client/src/interface.js [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+function sync(path, callback) {
+  if(!fs) {
+    return callback(new Error('Makedrive has not been initialized with a filesystem'));
+  }
+  if(!serverURI) {
+    return callback(new Error('Makedrive has not been initialized with a server URI'));
+  }
+  console.log('Coonection id: ', connectionId);
```
Comment (sedge) [inline client/src/interface.js]: delete

File: server/lib/filer-www.js [hunk 23]
```diff
@@ -177,6 +177,7 @@ function FilerWebServer(username) {
 FilerWebServer.prototype.handle = function(path, res) {
   var that = this;
   var fs = that.fs;
+  console.log(path);
```
Comment (sedge) [inline server/lib/filer-www.js]: delete

File: server/lib/socket-server.js [hunk 24]
```diff
@@ -10,34 +10,38 @@ module.exports = function( server ) {
...
+  wss.on('connection', function f(ws) {
+   ws.onmessage = function(data, flags) {
      // Capture the connectionId
      // Remove event listener for "this"
-     var match = /{"syncId"\s*:\s*"(\w{8}(-\w{4}){3}-\w{12}?)"}/.exec(data),
+     var match = /{"syncId"\s*:\s*"(\w{8}(-\w{4}){3}-\w{12}?)"}/.exec(data.data),
          // TODO: Research websocket authentication (so we can pass username here)
          sync;
 
+     console.log(data.data);
```
Comment (sedge) [inline server/lib/socket-server.js]: delete

File: server/lib/socket-server.js [hunk 25]
```diff
@@ -10,34 +10,38 @@ module.exports = function( server ) {
...
      if ( !match ) {
+       console.log('Not match');
```
Comment (sedge) [inline server/lib/socket-server.js]: delete

File: server/lib/socket-server.js [hunk 26]
```diff
@@ -10,34 +10,38 @@ module.exports = function( server ) {
...
+       if(!flags || (flags && !flags.binary)) {
+         data = data.data;
          try {
            data = JSON.parse(data);
-           Sync.messageHandler(data);
+           sync.messageHandler(data);
          } catch(error) {
            var Error = new SyncMessage(SyncMessage.RESPONSE, SyncMessage.ERROR);
            Error.setContent(error);
+           console.log(Error);
```
Comment (sedge) [inline server/lib/socket-server.js]: delete

File: server/lib/sync.js [hunk 27]
```diff
@@ -191,33 +205,41 @@ Sync.prototype.patch = function( diffs, callback ) {
...
-            diffs[i].diff[k].data = new Uint8Array(diffs[i].diff[k].data.__array);
+          if (diffs[i].diffs[k].data.__isUint8Array) {
+            diffs[i].diffs[k].data = new Uint8Array(diffs[i].diffs[k].data.__array);
           }
         }
       }
     }
   }
+  
+  console.log('Path: ', that.path);
```
Comment (sedge) [inline server/lib/sync.js]: delete

File: server/lib/sync.js [hunk 28]
```diff
@@ -256,49 +278,62 @@ Sync.prototype.setSrcList = function( srcList ){
 };
 
 Sync.prototype.messageHandler = function( data ) {
+  var that = this;
+  console.log(data);
```
Comment (sedge) [inline server/lib/sync.js]: delete

File: server/lib/sync.js [hunk 29]
```diff
@@ -256,49 +278,62 @@ Sync.prototype.setSrcList = function( srcList ){
...
   if(!data || !data.content) {
+    console.log('1');
```
Comment (sedge) [inline server/lib/sync.js]: delete

File: server/lib/sync.js [hunk 30]
```diff
@@ -256,49 +278,62 @@ Sync.prototype.setSrcList = function( srcList ){
...
     return this.socket.send(Sync.socket.errors.EUNDEF);
   }
-  if(!(data instanceof SyncMessage)) {
-    return this.socket.send(Sync.socket.errors.EINVAL);
-  }
+//  if(!(data instanceof SyncMessage)) {
```
Comment (sedge) [inline server/lib/sync.js]: What is this? Can it be deleted?

File: server/lib/sync.js [hunk 31]
```diff
@@ -256,49 +278,62 @@ Sync.prototype.setSrcList = function( srcList ){
...
+//  }
 
   var res;
 
   if(data.type === SyncMessage.REQUEST) {
 
     if(data.name === SyncMessage.SOURCE_LIST) {
-      if(this.socketState !== Sync.WSCON || this.socketState !== Sync.SRCLIST) {
+      if(this.socketState !== Sync.WSCON && this.socketState !== Sync.SRCLIST) {
+        console.log('3');
```
Comment (sedge) [inline server/lib/sync.js]: delete

File: server/lib/sync.js [hunk 32]
```diff
@@ -256,49 +278,62 @@ Sync.prototype.setSrcList = function( srcList ){
...
         return this.socket.send(Sync.socket.errors.ESTATE);
       }
-      rsync.sourceList(this.fs, this.path, rsyncOptions, function(err, srcList) {
+      return rsync.sourceList(this.fs, this.path, rsyncOptions, function(err, srcList) {
         if(err) {
+          console.log('4');
```
Comment (sedge) [inline server/lib/sync.js]: delete

File: server/lib/sync.js [hunk 33]
```diff
@@ -256,49 +278,62 @@ Sync.prototype.setSrcList = function( srcList ){
...
         }
-        return this.socket.send(JSON.stringify(res));
+        that.socket.send(JSON.stringify(res));
       });
     }
 
     if(data.name === SyncMessage.DIFF) {
-      if(this.socketState !== Sync.CHECKSUM || this.socketState !== Sync.DIFF) {
+      if(this.socketState !== Sync.CHECKSUM && this.socketState !== Sync.DIFF) {
+          console.log('5');
```
Comment (sedge) [inline server/lib/sync.js]: ALL THE LOGS! DELETE!

**Pull request merged**